cmd/argo/commands/get.go

```diff
@@ -116,25 +118,301 @@ func printWorkflowHelper(wf *wfv1.Workflow) {
+	// Used to store children which have a Non Boundary Parent from rendering perspective
+	// Maps non Boundary render Children name -> *nonBoundaryParentNode
+	nonBoundaryParentChildrenMap := make(map[string]*nonBoundaryParentNode)
+
+	// We have to do a 2 pass approach because anything that is a child
+	// of a nonBoundaryParent and also has a boundaryID we may not know which
+	// parent to attach to if we didn't see the nonBoundaryParent earlier
+	// in a 1 pass strategy
+
+	// 1st Pass Process enough of nonBoundaryParent nodes to know all their children
+	for id, status := range wf.Status.Nodes {
+		if isNonBoundaryParentNode(status.Type) {
+			n := nonBoundaryParentNode{node: id}
+			nonBoundaryParentMap[id] = &n
+
+			for _, child := range status.Children {
+				nonBoundaryParentChildrenMap[child] = &n
+			}
+		}
+	}
+
+	// 2nd Pass process everything
+	for id, status := range wf.Status.Nodes {
+		switch {
+		case isBoundaryNode(status.Type):
+			n := boundaryNode{node: id}
+			boundaryNodeMap[id] = &n
+			// Attach to my parent if needed
+			if attachToParent(wf, &n, nonBoundaryParentChildrenMap,
+				status.BoundaryID, boundaryNodeMap, parentBoundaryMap) {
+				renderTreeRoots[n.getID()] = &n
+			}
+			// Attach nodes who are in my boundary already seen before me to me
+			for _, val := range parentBoundaryMap[id] {
+				n.boundaryContained = insertSorted(wf, n.boundaryContained, val)
+			}
+		case isNonBoundaryParentNode(status.Type):
+			nPtr, ok := nonBoundaryParentMap[id]
+			if !ok {
+				log.Fatal("Unable to lookup node " + id)
+				return nil
+			}
+			// Attach to my parent if needed
+			if attachToParent(wf, nPtr, nonBoundaryParentChildrenMap,
+				status.BoundaryID, boundaryNodeMap, parentBoundaryMap) {
+				renderTreeRoots[nPtr.getID()] = nPtr
+			}
+			// All children attach directly to the nonBoundaryParents since they are already created
+			// in pass 1 so no need to do that here
+		case isExecutionNode(status.Type):
+			n := executionNode{node: id}
+			// Attach to my parent if needed
+			if attachToParent(wf, &n, nonBoundaryParentChildrenMap,
+				status.BoundaryID, boundaryNodeMap, parentBoundaryMap) {
+				renderTreeRoots[n.getID()] = &n
+			}
+			// Execution nodes don't have children
+		}
+	}
+
+	return renderTreeRoots
+}
+
+func (nodeInfo *boundaryNode) getNodeStatus(wf *wfv1.Workflow) wfv1.NodeStatus {
+	return wf.Status.Nodes[nodeInfo.node]
+}
+
+func (nodeInfo *nonBoundaryParentNode) getNodeStatus(wf *wfv1.Workflow) wfv1.NodeStatus {
+	return wf.Status.Nodes[nodeInfo.node]
+}
+func (nodeInfo *executionNode) getNodeStatus(wf *wfv1.Workflow) wfv1.NodeStatus {
+	return wf.Status.Nodes[nodeInfo.node]
+}
+
+func filterNode(node wfv1.NodeStatus) (bool, bool) {
+	if node.Type == wfv1.NodeTypeRetry && len(node.Children) == 1 {
+		return true, false
+	} else if node.Type == wfv1.NodeTypeStepGroup {
+		return true, true
+	}
+	return false, false
+}
+
+func renameNode(wf *wfv1.Workflow, node wfv1.NodeStatus) string {
```

Comment:
Can we rename this to humanizeNodeName? Took me some time to understand the purpose of this.